crawl-ref/source/tags.cc

```diff
@@ -3917,10 +3919,23 @@ static void tag_read_you_dungeon(reader &th)
     for (int i = 0; i < count_p; i++)
     {
         place_info = unmarshallPlaceInfo(th);
-        ASSERT(!place_info.is_global());
+        if (place_info.is_global() && th.getMinorVersion() <= TAG_MINOR_DESOLATION_GLOBAL)
+        {
+            // This is to fix some crashing saves that didn't import
+            // correctly, where the desolation slot occasionally gets marked
+            // as global on conversion from pre-0.19 to post-0.19a. Is setting
+            // the place info for branch `i` really a good idea? There's
+            // actually no guarantee that the save order will match the
+            // current branch order; this is really more of a best guess.
+            // HOWEVER, in all the instances I have, this works, and the place
+            // order hasn't changed since the iterator was introduced. (advil)
```

Comment:
That sounds reasonable, I'll update it tonight.  The minor tag idea (which was PF's suggestion) also helps with this, because I know that the save order hasn't changed in the past.